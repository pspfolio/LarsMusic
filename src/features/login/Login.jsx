import React, { Component } from 'react';
import { backgroundImage, backgroundPlaceholder } from './loginConstants';
import ProgressiveBackground from 'common/components/progressiveBackground/ProgressiveBackground';
import styled from 'styled-components';
import spotifyWhiteIcon from 'assets/images/Spotify_White.png';

const LoginContainer = styled.article`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
  padding: 0 40px;
  text-align: center;

  @media (max-width: 375px) {
    padding: 0 24px;
  }
`;

const MainHeader = styled.h1`
  text-align: center;
  font-size: 3rem;
  color: #fff;
  margin: 32px 0;
`;

const LoginParagraph = styled.p`
  color: white;
  font-size: 1.5rem;
  letter-spacing: 0.75px;
  margin: 0 0 24px 0;
`;

const LoginButton = styled.a`
  background-color: #1db954;
  height: 50px;
  border-radius: 500px;
  padding: 8px 48px 6px;
  min-width: 160px;
  border-width: 0;
  letter-spacing: 1.5px;
  text-transform: uppercase;
  white-space: normal;
  color: white;
  transition-property: background-color;
  transition-duration: 0.3s;
  text-decoration: none;
  font-weight: 500;
  line-height: 50px;
  display: flex;
  align-items: center;
  margin-top: 32px;
  justify-content: center;
  &:hover {
    background-color: #168d40;
  }
`;

const SpotifyIcon = styled.img`
  height: 30px;
  position: relative;
  padding: 0 8px;
`;

class Login extends Component {
  render() {
    return (
      <ProgressiveBackground background={backgroundImage} placeholder={backgroundPlaceholder}>
        <LoginContainer>
          <MainHeader>LarsMusic</MainHeader>
          <LoginParagraph>Track your album and vinyl collection and use Spotify data to help it</LoginParagraph>
          <LoginParagraph>Please login to Spotify and start use the application</LoginParagraph>
          <LoginButton href="http://localhost:8888/login">
            Log in <SpotifyIcon src={spotifyWhiteIcon} alt="logo of Spotify" />
          </LoginButton>
        </LoginContainer>
      </ProgressiveBackground>
    );
  }
}

export default Login;
