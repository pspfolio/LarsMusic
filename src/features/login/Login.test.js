import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Login from './Login';

describe('Login component', () => {
  it('renders without crashing', () => {
    shallow(<Login />);
  });

  it('Login renders correctly', () => {
    const tree = renderer.create(shallow(<Login />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
