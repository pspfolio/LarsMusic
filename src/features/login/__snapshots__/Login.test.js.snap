// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component Login renders correctly 1`] = `
<div
  className="sc-bdVaJa tHcNi"
>
  <article
    className="sc-bwzfXH irXouN"
  >
    <h1
      className="sc-htpNat iwPzKF"
    >
      LarsMusic
    </h1>
    <p
      className="sc-bxivhb fgjHCA"
    >
      Track your album and vinyl collection and use Spotify data to help it
    </p>
    <p
      className="sc-bxivhb fgjHCA"
    >
      Please login to Spotify and start use the application
    </p>
    <a
      className="sc-ifAKCX bwrGw"
      href="http://localhost:8888/login"
    >
      Log in 
      <img
        alt="logo of Spotify"
        className="sc-EHOje iupfgb"
        src="Spotify_White.png"
      />
    </a>
  </article>
</div>
`;
